006-strimzi-ui.md

```diff
@@ -0,0 +1,178 @@
+
+Finally, one capability which may be of interest (which may have bearing on how this is surfaced in Strimzi itself) is having a single UI, which can manage multiple Kafka clusters deployed via the Strimzi operator.
+
+In all of these cases, capabilities can be added in a prioritised order, and should be added in a phased manner themselves (for example, add a view of the brokers and their configuration, then the ability to modify select configuration).
+
+#### Proposed deployment
+
+This UI could be deployed as a part of Strimzi as follows:
+
+![Suggested deployment](./images/006-deployment.png)
+
+Where:
+
+- The UI is deployed standalone, akin to how the Kafka bridge, Kafka Connect are currently for example (so a new CRD would be required, with configuration needed to reference which cluster(s) to interact with).
+- The UI's spec would contain 0 to N Kafka clusters the UI will operate against, which will: 
+  - Contain 1 or more 'backend' entries. These represent backend sources of data which can be surfaced in the UI.
+    - One of these entries will be the `strimzi-api` (ie each Kafka kind will contain its own api server).
+  - Each of these entries will define items such as the address to use to connect to it, any auth credentials required, Transport security required, etc.
+- The UI's spec will contain general configuration for the UI - such as certificates to expose to clients etc.
+
+This could look as follows in a CR configured by a user (note that common fields such as image, or readiness and liveness probes have been omitted here for conciseness, but would be present in a full CR). Comments are included inline for clarification of intent/type:
+
+```
+...
+spec:
+  clusters: # Array/list
+  - name: 'dev-cluster' # string - metadata, but could be used in the UI (eg for cluster selection if more than one cluster) in the clusters array (order of which respected by the UI)
+    clientCert: <mounted TLS cert to serve to clients> # optional - if omitted, UI server will serve via http rather than https
+    uiConfig: <stringified JSON, containing general UI config - eg features to disable/enable, logging> # string - optional - if provided will override default configuration values
+    authAuthzMechanism: KafkaUser # string (although perhaps a custom type may be better). 'none' to mean disabled (ie no authentication or authorisation) - see assumptions below.
+    useAuthentication: true # boolean - only valid if authAuthzMechanism is not 'none'. Should the UI check a user's identity - see assumptions below 
+    useAuthorization: true # boolean - only valid if useAuthentication is true - should resources be protected/operated on by authoised users only. If false all users can access/modify all resources (eg topics) - see assumptions below 
```

Comment:
are these three fields related to an authn/authz mechanism between the UI server and cluster API backend or between a client (browser) and the UI server?

Reply:
When I wrote it, I was thinking between User and UI only. But you are right, this really could/should be for the whole stack (UI, api). As per my comment [here](https://github.com/strimzi/proposals/pull/6#issuecomment-658889455), it is probably worth while working this/aligning aspect across both proposals. I see it is a topic in today's community call as well - it would be good to discuss and explore options for this going forward more broadly.